fe1-web/src/core/objects/Hash.ts

```diff
@@ -44,6 +45,25 @@ export class Hash extends Base64UrlData {
     return this.toString() === o.toString();
   }
 
+  /**
+   * Creates a hash of a public key respecting the digital cash specification
+   * https://github.com/dedis/popstellar/blob/42e18f176a91371f28653da8815d9591c38f2ee6/be2-scala/src/main/scala/ch/epfl/pop/model/objects/PublicKey.scala#L8
+   *
+   * @param publicKey the public key to hash
+   */
+  public static fromPublicKey(publicKey: string | PublicKey): Hash {
+    const hash = sha256.create();
+
+    // We need the raw public key
+    const decodedPK = Base64UrlData.fromBase64(publicKey.valueOf()).toBuffer();
+
+    const bString = hash.update(decodedPK).array();
+    // Taking only the first 20 bytes
+    const str = String.fromCharCode(...bString.slice(0, 20));
+
+    return new Hash(Base64UrlData.encode(str, 'binary').valueOf());
```

Comment:
If it works it works, but it seems a bit convoluted and fragile to stuff your bytes into a (UTF-16) string and then re-decode it as "binary". If we could go through `Buffer` instead.

Reply:
True and done